api/src/routes/userOrganisationSettings/router-test.js

```diff
@@ -0,0 +1,367 @@
+        org2OrganisationSetting,
+      ]);
+
+      await apiApp
+        .patch(`/v2/users/${targetUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${userManagerOrg1Token}`)
+        .send({ roles: [nonUserManagerRoleId, userManagerRoleId] })
+        .expect(200);
+
+      const updatedTargetUser = await User.findOne({ _id: targetUser._id });
+      assert.equal(updatedTargetUser.organisationSettings.length, 2);
+      assert.equal(updatedTargetUser.organisationSettings[0].roles.length, 2);
+      assert.equal(updatedTargetUser.organisationSettings[0].roles[1].toString(), userManagerRoleId);
+    });
+
+    it('can not update an organisationSettings if body includes invalid fields', async () => {
+      const targetUser = await createTargetUser([
+        org1OrganisationSetting,
+        org2OrganisationSetting,
+      ]);
+
+      await apiApp
+        .patch(`/v2/users/${targetUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${userManagerOrg1Token}`)
+        .send({ timezone: 'Asia/Tokyo' })
+        .expect(400);
+    });
+
+    it('can delete an organisationSettings of org1 for anyone', async () => {
+      const targetUser = await createTargetUser();
+
+      await apiApp
+        .delete(`/v2/users/${targetUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${userManagerOrg1Token}`)
+        .expect(200);
+    });
+  });
+
+  describe('a user who does not have manage users', () => {
+    let nonUserManagerOrg1Token;
+    let loginUser;
+
+    beforeEach(async () => {
+      const organisationSettings = [{
+        organisation: org1Id,
+        roles: [nonUserManagerRoleId],
+      }];
+      loginUser = await createLoginUser([], organisationSettings);
+      nonUserManagerOrg1Token = await createOrgJWT(loginUser, org1Id, 'native');
+    });
+
+    it('can not create an organisationSettings for anyone', async () => {
+      const targetUser = await createTargetUser();
+
+      await apiApp
+        .post(`/v2/users/${targetUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${nonUserManagerOrg1Token}`)
+        .send({ todo: 'something' })
+        .expect(401);
+    });
+
+    it('can not create its own organisationSettings of org1', async () => {
+      await apiApp
+        .post(`/v2/users/${loginUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${nonUserManagerOrg1Token}`)
+        .send({ invalidField: 'x' })
+        .expect(401);
+    });
+
+    it('can update a its own organisationSettings of org1 if body has only updatable fields', async () => {
+      await apiApp
+        .patch(`/v2/users/${loginUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${nonUserManagerOrg1Token}`)
+        .send({ samlEnabled: false })
+        .expect(200);
+
+      // const updatedLoginUser = await User.findOne({ _id: loginUser._id });
```

Comment:
Is this a TODO because it needs to be done in Enterprise instead @h-kanazawa?

Reply:
Yes, the assert is for enterprise.